integration-tests/src/test/java/org/openmrs/maven/plugins/packager/config/ValidateContentPackageMojoTest.java

```diff
@@ -0,0 +1,95 @@
+package org.openmrs.maven.plugins.packager.config;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.FileInputStream;
+import java.util.Properties;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class ValidateContentPackageMojoTest {
+	
+	@InjectMocks
+	private ValidateContentPackageMojo mojo = new ValidateContentPackageMojo();
+	
+	@Mock
+	private FileInputStream mockFileInputStream;
+	
+	@Mock
+	private Properties mockProperties;
+	
+	@Rule
+	public ExpectedException exceptionRule = ExpectedException.none();
+	
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+	
+	public void executeValidContentPropertiesFileWithoutErrors() throws Exception {
+		// setup
+		String validPropertiesFile = "src/test/resources/config-test-child/valid-content.properties";
+		mojo.sourceFile = validPropertiesFile;
+		// replay
+		mojo.execute();
+		
+		// verify
+		// No exception should be thrown
+	}
+	
+	public void executeInvalidContentPropertiesFormatWithException() throws Exception {
+		// setup
+		String invalidPropertiesFile = "src/test/resources/config-test-child/invalid-content.properties";
+		mojo.sourceFile = invalidPropertiesFile;
+		
+		try {
+			mojo.execute();
+		}
+		catch (MojoExecutionException e) {
+			assertTrue("MojoExecutionException raised", true);
+		}
+		
+	}
+	
+	@Test
+	public void testInValidVersions() {
+		assertFalse(mojo.isValid("1.0.01"));
+		assertFalse(mojo.isValid("1.0.0-alpha@"));
+		assertFalse(mojo.isValid("1.0.0-1234-"));
+		assertFalse(mojo.isValid("abc"));
+		assertFalse(mojo.isValid("1..0"));
+		assertFalse(mojo.isValid("1.0.0.0"));
+		assertFalse(mojo.isValid("latest"));
+		assertFalse(mojo.isValid("next"));
+	}
+	
+	@Test
+	public void testValidVersionRanges() {
+		//java_semver treats it as valid, short for 1.0.0, even though patch# is missing
+		assertTrue(mojo.isValid("1.0"));
+		assertTrue(mojo.isValid("0.13.0"));
+		assertTrue(mojo.isValid("^2.13.0"));
+		assertTrue(mojo.isValid("~0.13.0"));
+		assertTrue(mojo.isValid(">0.13.0"));
+		assertTrue(mojo.isValid("<3.0.0"));
+		assertTrue(mojo.isValid(">=3.0.0"));
+		assertTrue(mojo.isValid("<=3.0.0"));
+		assertTrue(mojo.isValid("1.0.0 - 1.10.10"));
+		assertTrue(mojo.isValid("<2.1.0 || >2.6.0"));
+		//	assertTrue(mojo.isValid(">=1.0.0-SNAPSHOT"));
```

Comment:
Does this not pass validation? This would be a bit of a show-stopper if it doesn't.

Reply:
It doesn't automatically pass validation, but we can handle this scenario, Range expression >=1.0.0-SNAPSHOT is not considered valid for snapshot versions as they are typically treated as pre-release versions and range expressions are not commonly used with them. Do you want to handle range expressions to SNAPSHOT versions as well?

Reply:
For clarification, 1.0.0-SNAPSHOT is a valid SemVer expression, but not a valid range (as explained earlier). For example, specifying =1.0.0-SNAPSHOT is valid, but >= 1.0.0-SNAPSHOT is not. This brings me back to my question from a few days ago: for omod, owa, and spa, do ranges need to be valid range expressions, or is it sufficient if they are valid SemVer expressions? or do you want to handle pre-release versions for ranges?

Reply:
We can update our current implementation to handle pre-release versions in ranges. I plan to strip the pre-release version (e.g., remove '-SNAPSHOT' from '>= 1.0.0-SNAPSHOT') and then validate the range. Do you see any concerns or issues with this approach?

Reply:
> I guess we need to either look for a library that handles this as well or create our own implementation.

Which would you prefer?

Reply:
/**
	 * Validates whether a given value is a valid SemVer expression or range.
	 *
	 * @param value the value to validate
	 * @return true if the value is a valid SemVer expression or range, false otherwise
	 */
	protected boolean isValid(String value) {
		try {
			// Attempt to parse the value as a range expression
			ExpressionParser.newInstance().parse(value);
			return true;
		}
		catch (Exception e) {
			// If parsing as a range fails, try removing range operators and check if its a valid version
			try {
				String rangeOperatorsRegex = "^[<>=~^\\s*]+";
				// Check if the value contains range operators
				if (value.matches(rangeOperatorsRegex + ".*")) {
					// Remove range operators if present
					value = value.replaceAll(rangeOperatorsRegex, "").trim();
					
				}
				//check if the value is a valid version
				Version.parse(value);
				return true;
			}
			catch (Exception ex) {
				// If parsing as both a range and a single version fails, return false
				return false;
			}
		}
	}
```

Reply:
This class is responsible for checking if the values are valid SemVer (versions/ranges), correct?

We have two options:

**Check for a Different Library**: However, since this library is used elsewhere in our project, let's try to make it work.

**Customize Our Implementation**: Given that this library doesn't support ranges with pre-release versions, we have the following options:

1. Remove Everything After Major, Minor, and Patch: Validate the version against the range.

2. Remove the Range Operators: Validate it as a version. The example below illustrates this.

This plugin only validates the content.properties; the SDK handles downloading artifacts. Meanwhile, I'll check if the Maven artifact resolver can handle ranges with SNAPSHOT versions. Can you let me know what you think?

Reply:
```
 /** Validates whether a given value is a valid SemVer expression or range.
 *
 * @param value the value to validate
 * @return true if the value is a valid SemVer expression or range, false otherwise
 */
protected boolean isValid(String value) {
	try {
		// Attempt to parse the value as a range expression
		ExpressionParser.newInstance().parse(value);
		return true;
	}
	catch (Exception e) {
		// If parsing as a range fails, try removing range operators and check if its a valid version
		try {
			String rangeOperatorsRegex = "^[<>=~^\\s*]+";
			// Check if the value contains range operators
			if (value.matches(rangeOperatorsRegex + ".*")) {
				// Remove range operators if present
				value = value.replaceAll(rangeOperatorsRegex, "").trim();
				
			}
			//check if the value is a valid version
			Version.parse(value);
			return true;
		}
		catch (Exception ex) {
			// If parsing as both a range and a single version fails, return false
			return false;
		}
	}
}
		
```

Reply:
was Reading about how Maven handles SNAPSHOT version ranges, Based on that understanding, i propose, we update the values in content.properties on a case by case basis if they include SNAPSHOT versions in ranges, 
for ranges like >= 1.0.0-SNAPSHOT, we can update the value to <1.0.0-SNAPSHOT
for ranges like <=1.0.0-SNAPSHOT (is it required?) we can update it to < 1.0.0
for the time being, we can raise an error for other range operators. Out of the 3, i think this will work.